### ISSUE TITLE
Qemu commandline arg support

### ISSUE THREAD

**Issue by waveywaves** [kubevirt/kubevirt#3745]
**Is this a BUG REPORT or FEATURE REQUEST?**: Feature Request
 /kind enhancement

**What happened**:
Not able to provide custom qemu commandline args via the VirtualMachine CR

**What you expected to happen**:
Be able to provide custom qemu commandline args via the VirtualMachine CR

**How to reproduce it (as minimally and precisely as possible)**:
-

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
```
Client Version: version.Info{GitVersion:"v0.30.0", GitCommit:"736b4aec6b2e92f2ab09ccfa6c5eb79366e88e5a", GitTreeState:"clean", BuildDate:"2020-06-04T14:50:54Z", GoVersion:"go1.12.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"{gitVersion}", GitCommit:"{gitCommit}", GitTreeState:"{gitTreeState}", BuildDate:"{buildDate}", GoVersion:"go1.12.8", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.1+3f6f40d", GitCommit:"3f6f40d", GitTreeState:"clean", BuildDate:"2020-06-08T07:13:25Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by waveywaves** [comment]
QEMUCmd Struct seems to get this job done in case of some network configs I saw around. Any chances on mapping that directly to the domain.Spec ?

**Comment by rmohr** [comment]
@waveywaves we have a sidecar-hook solution in place which allows you to modify the domain xml in arbitrary ways. Have a look at https://github.com/kubevirt/kubevirt/tree/master/cmd/example-hook-sidecar for an example. Hope that helps.

**Comment by dhruvik7** [comment]
Is the sidecar hook still the preferred solution here? @rmohr

**Comment by victortoso** [comment]
hi @dhruvik7, yes, there is an issue open for it #8420 and a PR quite close to get merged too #9166
